gr-fec/include/gnuradio/fec/viterbi.h

```diff
@@ -24,30 +24,40 @@
  * But it fits so nicely into a 32-bit machine word...
  */
 
+#ifndef INCLUDED_VITERBI_H
+#define INCLUDED_VITERBI_H
+
 #include <gnuradio/fec/api.h>
 
-struct viterbi_state {
-  unsigned long path;	/* Decoded path to this state */
-  long metric;		/* Cumulative metric to this state */
-};
+namespace gr {
+  namespace fec {
+
+    struct FEC_API viterbi_state {
+      unsigned long path;   /* Decoded path to this state */
```

Comment:
Reading the "it fits so nicely into a 32-bit word" comment above: this should explicitly be an `uint32_t`.

Reply:
actually, since we have him, could @SpectreJan comment on what the appropriate type sizes would be here.

Reply:
What I take away, `sizeof(metric)` should be `sizeof(**mettab)`, and reading decode.cc:

<pre><font color="#4E9A06"><b>lib/viterbi/decode.cc</b></font>
<font color="#C4A000"><b>48</b></font>:  int <span style="background-color:#C4A000"><font color="#2E3436">amp</font></span> = 100;
<font color="#C4A000"><b>52</b></font>:  gen_met(mettab, <span style="background-color:#C4A000"><font color="#2E3436">amp</font></span>, esn0, 0.0, 4);
<font color="#C4A000"><b>67</b></font>:      unsigned char sym = syms[i] &gt; 0 ? -<span style="background-color:#C4A000"><font color="#2E3436">amp</font></span> : <span style="background-color:#C4A000"><font color="#2E3436">amp</font></span>;
</pre>

I'd argue that `amp` should be uchar (or, while we're at it, `uint8_t`), generally.

Reply:
I would need to get back into the algorithm and implementation. At first glance uint32_t makes sense, since everything seems to be hardcoded for the 32bit words. But at this point I am not sure enough to make recommendation.

Reply:
As for the second...yes uchar makes much more sense here. But seein as line 67 has a fixme comment. Maybe we can get two birds with one stone?